### ISSUE TITLE
Verssion 2.8.2 - Error when saving configuration

### ISSUE THREAD

**Issue by delzear** [mrlt8/docker-wyze-bridge#1169]
Failed to save add-on configuration, invalid value or type (must have a partial ordering).
On Home Assistant Add On
The error disappears when I remove the API_ID and API_KEY fields but the addon won't start.

**Comment by poiromaniax** [comment]
Same

**Comment by mrlt8** [comment]
Thanks for the feedback, can you see if it works now?

**Comment by poiromaniax** [comment]
@mrlt8 Is there supposed to be an update to the addon? Dont see one showing...

Still not working for me

**Comment by mrlt8** [comment]
@poiromaniax can you try going to the add-on store and clicking the three dots near the top -> "Check for updates".

**Comment by poiromaniax** [comment]
@mrlt8 nothing

**Comment by delzear** [comment]
Works great! Thanks!

**Comment by baseer216** [comment]
I'm having the same issue as OP

Failed to save add-on configuration, Missing required option 'API_ID' in Docker Wyze Bridge (7094bb28_docker_wyze_bridge). Got {'WYZE_EMAIL': 'redacted', 'WYZE_PASSWORD': 'redacted!', 'API_ID': None, 'API_KEY': None, 'NET_MODE': 'ANY', 'SNAPSHOT': 'Disable', 'MQTT_DTOPIC': 'homeassistant', 'ENABLE_AUDIO': True, 'MOTION_API': False, 'ON_DEMAND': True, 'SUBSTREAM': False, 'CAM_OPTIONS': []}

I've installed Verssion 2.8.3

**Comment by cevans3505** [comment]
Same issue after update. Does not seem to accept saving the configuration if API ID and API key are omitted despite supposedly being optional.

**Comment by farrc4780** [comment]
Same issue here as well.

**Comment by dcw88** [comment]
Same error even with the latest version 2.8.3.  It kept saying API_ID is missing from the configuration and Docker Wyze Bridge won't start.

**Comment by felafel** [comment]
I get this when trying to start ....
Missing required option 'API_ID' in Docker Wyze Bridge (7094bb28_docker_wyze_bridge).

This has not been working for me for over a year so who knows.  It used to refresh manually at times but now I just update hoping it will work.

**Comment by TheRealWaldo** [comment]
The same happens here; I updated it to the latest version and still encountered the error.

I'd gladly fill in the ID and Key if there were any documentation on where to obtain them.

**Comment by farrc4780** [comment]
I updated to the new version and it is working for me now

**Comment by delzear** [comment]
The documentation has been tagged to the top of the README.md file for some time.

https://support.wyze.com/hc/en-us/articles/16129834216731-Creating-an-API-Key

**Comment by baseer216** [comment]
> The documentation has been tagged to the top of the README.md file for some time.
> 
> https://support.wyze.com/hc/en-us/articles/16129834216731-Creating-an-API-Key

THANK YOU!!!!! this resolved my issue

**Comment by cevans3505** [comment]
Indeed, creating the API key and including it in the config does resolve the issue, but the issue remains that this is _supposed_ to be an optional value. Either needs to be made mandatory, or properly handling saving the config with no API key present.

**Comment by felafel** [comment]
Putting in the key id and api key worked for me; but I still have cameras that don't refresh (unavailable) unless done manually in the addon UI.  Set to 120 on the page but not sure if that means anything outside the page.  My floodlight pro is not found but that seems to be expected at this time.

MOTION_API -- any ideas on what this does?

Thanks in advance

**Comment by mrlt8** [comment]
@cevans3505 Unfortunately, Wyze switched off the v2 and v3 authentication endpoints #1163 a couple of days ago, and I'm still exploring other authentication methods at this time.


@felafel you can try setting the ON_DEMAND option in the config to False or adjust the SNAPSHOT option if you only need to update the snapshot.
